common/src/main/java/com/wynntils/models/containers/BankModel.java

```diff
@@ -0,0 +1,164 @@
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class BankModel extends Model {
+    @Persisted
+    private final Storage<Integer> finalBankPage = new Storage<>(21);
+
+    @Persisted
+    private final Storage<Integer> finalBlockBankPage = new Storage<>(12);
+
+    @Persisted
+    private final Storage<Integer> finalBookshelfPage = new Storage<>(10);
+
+    @Persisted
+    private final Storage<Integer> finalMiscBucketPage = new Storage<>(10);
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customBankPageNames = new Storage<>(new TreeMap<>());
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customBlockBankPageNames = new Storage<>(new TreeMap<>());
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customBookshelfPageNames = new Storage<>(new TreeMap<>());
+
+    @Persisted
+    private final Storage<Map<Integer, String>> customMiscBucketPageNames = new Storage<>(new TreeMap<>());
+
+    private boolean editingName;
+    private int currentPage = 1;
+    private SearchableContainerType currentContainer;
+
+    public BankModel(ContainerModel containerModel) {
+        super(List.of(containerModel));
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onScreenInit(ScreenInitEvent e) {
+        if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
+
+        if (Models.Container.isBankScreen(screen)) {
+            currentContainer = SearchableContainerType.BANK;
+        } else if (Models.Container.isBlockBankScreen(screen)) {
+            currentContainer = SearchableContainerType.BLOCK_BANK;
+        } else if (Models.Container.isBookshelfScreen(screen)) {
+            currentContainer = SearchableContainerType.BOOKSHELF;
+        } else if (Models.Container.isMiscBucketScreen(screen)) {
+            currentContainer = SearchableContainerType.MISC_BUCKET;
+        } else {
+            currentContainer = null;
+            currentPage = 1;
+            return;
+        }
+
+        currentPage = Models.Container.getCurrentBankPage(screen);
+
+        editingName = false;
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onScreenClose(ScreenClosedEvent e) {
+        currentContainer = null;
+        currentPage = 1;
+        editingName = false;
+    }
+
+    public Optional<String> getPageName(int page) {
+        String customPageName =
+                switch (currentContainer) {
+                    case BANK -> customBankPageNames.get().get(page);
+                    case BLOCK_BANK -> customBlockBankPageNames.get().get(page);
+                    case BOOKSHELF -> customBookshelfPageNames.get().get(page);
+                    case MISC_BUCKET -> customMiscBucketPageNames.get().get(page);
+                    default -> null;
+                };
+
+        return Optional.ofNullable(customPageName);
+    }
+
+    public void saveCurrentPageName(String nameToSet) {
```

Comment:
Should I add .touched to each of these or is it not a good idea to call that a lot?